src/OrchardCore/OrchardCore.DisplayManagement.Liquid/HtmlContentValue.cs

```diff
@@ -57,7 +58,7 @@ public override string ToStringValue()
 
         public override void WriteTo(TextWriter writer, TextEncoder encoder, CultureInfo cultureInfo)
         {
-            _value.WriteTo(writer, HtmlEncoder.Default);
+            _value.WriteTo(writer, NullHtmlEncoder.Default);
```

Comment:
We need to double-check this one, where is this HtmlContentValue used.

Reply:
I will check this one.

Reply:
@sebastienros

So, it is only used for liquid in the `ShapeStringify` and `ShapeRender` filters that use the `displayHelper` to execute shapes. It is a wrapper holding an `IHtmlContent` that may be a simple `HtmlString` or e.g a `ViewBuffer` composed of `string` / `HtmlString` and others `IHtmlContent` that need to be encoded or not depending on their implementation.

So, i tried the following with some special charaters in an article Subtitle.

    {{ Model.Content["Article-Subtitle"] | shape_render }}

This with the `HtmlEncoder` or the `NullHtmlEncoder`

And mixed with these 2 cases in `TextField.cshtml`

    @Model.Text
    @(new StringHtmlContent(Model.Text))

With `@Model.Text` the result is the same with or without the `HtmlEncoder`. We retrieve it above as an already encoded string. Note: with `@Html.Raw()` we retrieve it as an `HtmlString` => idem same results but not encoded.

    @Model.Text
    _value.WriteTo(writer, (HtmlEncoder)encoder);
     This is what I do. Special characters &gt; &#xE9;&#xE9;&#xE9;

    @Model.Text
     _value.WriteTo(writer, NullHtmlEncoder.Default);
    This is what I do. Special characters &gt; &#xE9;&#xE9;&#xE9;

But with `@(new StringHtmlContent(Model.Text))` the results are different.

    @(new StringHtmlContent(Model.Text))
    _value.WriteTo(writer, NullHtmlEncoder.Default);
    This is what I do. Special characters > ééé

    @(new StringHtmlContent(Model.Text))
    _value.WriteTo(writer, (HtmlEncoder)encoder);
    This is what I do. Special characters &gt; &#xE9;&#xE9;&#xE9;

My conclusion would be that it is better to use `(HtmlEncoder)encoder`.